src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt

```diff
@@ -152,28 +150,39 @@ class Cargo(private val cargoExecutable: Path) {
             }
 
             if (settings.compileAllTargets && checkSupportForBuildCheckAllTargets()) add("--all-targets")
-            if (settings.useOffline) add("-Zoffline")
             addAll(ParametersListUtil.parse(settings.externalLinterArguments))
         }
 
         val useClippy = settings.externalLinter == ExternalLinter.CLIPPY
             && !checkNeedInstallClippy(project, cargoProjectDirectory)
         val checkCommand = if (useClippy) "clippy" else "check"
-        return CargoCommandLine(checkCommand, cargoProjectDirectory, arguments).execute(owner, ignoreExitCode = true)
+        return CargoCommandLine(checkCommand, cargoProjectDirectory, arguments)
+            .execute(project, owner, ignoreExitCode = true)
     }
 
-    fun toColoredCommandLine(commandLine: CargoCommandLine): GeneralCommandLine =
-        toGeneralCommandLine(commandLine, colors = true)
+    fun toColoredCommandLine(project: Project, commandLine: CargoCommandLine): GeneralCommandLine =
+        toGeneralCommandLine(project, commandLine, colors = true)
 
-    fun toGeneralCommandLine(commandLine: CargoCommandLine): GeneralCommandLine =
-        toGeneralCommandLine(commandLine, colors = false)
+    fun toGeneralCommandLine(project: Project, commandLine: CargoCommandLine): GeneralCommandLine =
+        toGeneralCommandLine(project, commandLine, colors = false)
 
-    private fun toGeneralCommandLine(commandLine: CargoCommandLine, colors: Boolean): GeneralCommandLine =
+    private fun toGeneralCommandLine(project: Project, commandLine: CargoCommandLine, colors: Boolean): GeneralCommandLine =
         with(patchArgs(commandLine, colors)) {
             val parameters = buildList<String> {
                 if (channel != RustChannel.DEFAULT) {
                     add("+$channel")
                 }
+                if (project.rustSettings.useOffline) {
+                    val cargoProject = findCargoProject(project, additionalArguments, workingDirectory)
+                    val rustcVersion = cargoProject?.rustcInfo?.version?.semver
+                    if (rustcVersion != null) {
```

Comment:
Let's move offline flag addition after `add(command)` line to avoid unusual order of arguments
At this moment: 
`cargo --offline build --color=always --all --all-targets --message-format=json`
Expected:
`cargo build --offline --color=always --all --all-targets --message-format=json`
of even:
`cargo build --color=always --all --all-targets --message-format=json --offline`